src/detect-content.c

```diff
@@ -435,7 +438,14 @@ int DetectContentParseTest01 (void)
     char *teststring = "\"abc\\:def\"";
     char *teststringparsed = "abc:def";
 
-    cd = DetectContentParse(teststring);
+    uint16_t spm_matcher = SinglePatternMatchDefaultMatcher();
+    SpmThreadCtx *spm_thread_ctx = SpmInitThreadCtx(spm_matcher);
+    if (spm_thread_ctx == NULL) {
```

Comment:
can you replace this by the new FAIL_IF(expr) logic? See for example https://github.com/inliniac/suricata/commit/7f700a137cd54497c16d99a0c4418f1296a87aaa